### DISCUSSION TITLE
Row Selection - Delete Rows based on SelectedRowIds

### DISCUSSION THREAD

**Discussion by tdavis3** [TanStack/table#2027] [General]
Hi,
I.m having trouble deleting the selected rows in my table.  Not sure how to filter my data based on the selectedRowIds from the table state.  My data is a list of objects each containing a unique alphanumeric value.  Is it possible to use that as the row ids?

**Comment by fnn** [comment]
Yes. Use the getRowId table option to set the ID for each row.
`getRowId: React.useCallback(row => row.id, [])`
